Add recipe for the xNVMe system library and CLI

safl opened the pull request:
This is useful in itself and also serves as a preparation for a Conda package for the xNVMe Python bindings, making the library available for the xNVMe Python module.

<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| c/c++           | `@conda-forge/help-c-cpp`     |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

safl commented:
I am willing to be a maintainer of this Conda package.

safl commented:
@conda-forge/staged-recipes this is ready for review, I have a question in PR message which, if possible can expand the package to include OSX.
@conda-forge/help-c-cpp

Looking forward to your input!

xhochy commented on recipes/xnvme-sys/meta.yaml:
```diff
@@ -0,0 +1,74 @@
+{% set sha256 = "6d42d0bd7e6b395a37869a8713d138d332500f20ecc272ebf6789471026a7191" %}
+
+package:
+  name: {{ name|lower }}-sys
```
What's the reason in adding `sys` here?

safl replied:
The intent was to provide this package for the system-libraries and tools that **xNVMe** provides and then a package named **xnvme** for the Python bindings. However, it probably makes better sense for this package to be **xnvme** and then the other package can be **xnvme-py**, or include it in **xnvme** package? What is the convention to follow here?

For now, then i'll rename it to **xnvme** instead of **xnvme-sys**.

xhochy replied:
That makes sense. Name it libxvme instead. That is our usual convention for these kind of packages.

safl replied:
ack.

xhochy commented on recipes/xnvme-sys/meta.yaml:
```diff
@@ -0,0 +1,74 @@
+---
```
```suggestion
```

xhochy commented on recipes/xnvme-sys/meta.yaml:
```diff
@@ -0,0 +1,74 @@
+
+source:
+  url: https://github.com/{{ name }}/{{ name }}/releases/download/v{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Please put the sha256 directly here

safl replied:
ack.

xhochy commented on recipes/xnvme-sys/meta.yaml:
```diff
@@ -0,0 +1,74 @@
+
+build:
+  skip: true  # [not linux]
+  script: |
```
For better readbility, put this into `build.sh`

safl replied:
ack. I also put the tests into run_test.sh.

xhochy commented on recipes/xnvme-sys/meta.yaml:
```diff
@@ -0,0 +1,74 @@
+    mkdir -p $PREFIX/etc/conda/deactivate.d
+
+    # Activation script
+    echo 'export PATH=$PREFIX/bin:$PREFIX/usr/bin:$PREFIX/usr/local/bin:$PREFIX/usr/local/lib/x86_64-linux-gnu:$PATH' > $PREFIX/etc/conda/activate.d/xnvme.sh
```
Please ensure that the binaries are installed in typical conda paths. You should not need to adjust PATH with activation scripts.

safl replied:
This is giving me some challenges when using ``pkg-config``. From what I can tell then the ``build.sh`` does as other meson-projects do:

```
meson setup builddir --prefix=$PREFIX
```

However, the command in ``run_test.sh`` which checks whether ``pkg-config`` works, it fails unless these things are setup in the activation / de-activation scripts. It would be great with a pointer on how this should otherwise be done.

xhochy replied:
You probably need to pass `$MESON_ARGS` to meson to get them right.

safl replied:
Thanks @xhochy ! This is the thing that was missing.

safl commented:
Hi @xhochy,

Thanks for your review! I've addressed all the points except the last one about the activation/deactivation scripts. I need to experiment a bit more to figure out how to handle this. If you have any pointers or suggestions, they would be greatly appreciated.

xhochy commented on recipes/libxnvme/meta.yaml:
```diff
@@ -0,0 +1,50 @@
+---
```
Please use the same formatting as other recipes, e.g. remove this line.

safl replied:
ack

xhochy commented on recipes/libxnvme/meta.yaml:
```diff
@@ -0,0 +1,50 @@
+  - pkg-config
+
+  host:
+  - {{ stdlib('c') }}
```
```suggestion
```

safl replied:
ack

xhochy commented on recipes/libxnvme/meta.yaml:
```diff
@@ -0,0 +1,50 @@
+  - {{ stdlib('c') }}
+  - libaio  # [linux]
+  - liburing  # [linux]
+  - pkg-config
```
```suggestion
```

safl replied:
ack

xhochy commented on recipes/libxnvme/meta.yaml:
```diff
@@ -0,0 +1,50 @@
+  - liburing  # [linux]
+  - pkg-config
+
+  run:
+  - {{ stdlib('c') }}
+  - libaio  # [linux]
+  - liburing  # [linux]
+  - pkg-config
```
```suggestion
```

safl replied:
ack

xhochy commented on recipes/libxnvme/build.sh:
```diff
@@ -0,0 +1,16 @@
+mkdir -p $PREFIX/etc/conda/activate.d
+mkdir -p $PREFIX/etc/conda/deactivate.d
+
+# Activation script
```
These scripts need to be removed, we won't merge this as long as it doesn't use the normal path like $PREFIX/bin

safl replied:
ack. In case anybody else stumbles over this, then these scripts, that I thought were needed, are not, since the ``$MESON_ARGS`` as @xhochy suggested sets, among other things  ``-Dlib=...`` and ``--prefix`` which is what is needed for executables, and the pkg-config xnvme.pc to be located correctly.

safl commented:
Hi @xhochy,

Thank you for your help! Your suggestion to use ``MESON_ARGS`` was incredibly useful. It was the key detail I had overlooked, which led to creating those hacky activation/deactivation scripts. I have now removed those scripts and addressed your comments. The changes have been squashed into the first commit.

Thanks again!
